ts/features/bonus/cgn/components/merchants/CgnMerchantsFilters.tsx

```diff
@@ -0,0 +1,201 @@
+    );
+    Keyboard.dismiss();
+  };
+
+  const onOrderSelect = (value: string) => {
+    setSelectedOrder(value);
+    Keyboard.dismiss();
+  };
+
+  const onRemoveButton = () => {
+    setSearchValue("");
+    setSelectedOrder(orders.distance.value);
+    setCheckedCategories([]);
+    props.onClose();
+  };
+
+  const selectedFilters =
+    checkedCategories.length + (searchValue.length > 0 ? 1 : 0);
+
+  return (
+    <Container>
+      <AppHeader>
+        <Left />
+        <Body style={{ alignItems: "center" }}>
+          <H5 weight={"SemiBold"} color={"bluegrey"}>
+            {I18n.t("bonus.cgn.merchantsList.filter.title")}
+          </H5>
+        </Body>
+        <Right>
+          <ButtonDefaultOpacity onPress={props.onClose} transparent={true}>
+            <IconFont name="io-close" />
+          </ButtonDefaultOpacity>
+        </Right>
+      </AppHeader>
+      <SafeAreaView style={IOStyles.flex}>
+        <ScrollView
+          style={[IOStyles.flex]}
+          bounces={false}
+          keyboardShouldPersistTaps={"handled"}
+        >
+          <View style={IOStyles.horizontalContentPadding}>
+            <H2>{I18n.t("bonus.cgn.merchantsList.filter.searchTitle")}</H2>
+            <Item>
+              <Input
+                value={searchValue}
+                onChangeText={setSearchValue}
+                placeholderTextColor={IOColors.bluegreyLight}
+                placeholder={I18n.t(
+                  "bonus.cgn.merchantsList.filter.inputPlaceholder"
+                )}
+              />
+              <IconFont name="io-search" color={IOColors.bluegrey} />
+            </Item>
+            <View spacer large />
+            {props.isLocal && (
+              <>
+                <H2>{I18n.t("bonus.cgn.merchantsList.filter.addressTitle")}</H2>
+                <Item>
+                  <Input
+                    value={address}
+                    onChangeText={setAddress}
+                    placeholderTextColor={IOColors.bluegreyLight}
+                    placeholder={I18n.t(
+                      "bonus.cgn.merchantsList.filter.addressPlaceholder"
+                    )}
+                  />
+                  <IconFont name="io-place" color={IOColors.bluegrey} />
+                </Item>
+                <View spacer />
+                <DistanceSlider
+                  value={distance}
+                  onValueChange={setDistance}
+                  disabled={address.length === 0}
+                />
+                <View spacer large />
+              </>
+            )}
+            <H2>{I18n.t("bonus.cgn.merchantsList.filter.categories")}</H2>
+            <View spacer small />
+            <FlatList
+              data={categories}
+              keyExtractor={cat => cat.type}
+              ItemSeparatorComponent={() => (
+                <ItemSeparatorComponent noPadded={true} />
+              )}
+              renderItem={(listItem: ListRenderItemInfo<Category>) => (
```

Comment:
Could it be defined in a dedicated function?
It would keep the code more readable

Reply:
solved in 56ddec7